### PR TITLE
Jetpack Cloud: Create and organize routes, sections, and root page components

### PR DESCRIPTION
**Pull request by rcoll**
#### Changes proposed in this Pull Request

* Created appropriate routes for each of the pages that we'll have initially
* Reorganized page-level components into `sections` folder
* Connected page-level components to app state
* General cleanup of root controller (more to come)

#### Testing instructions

Visit the following pages, with your own jetpack test site slug substituted. Each should load its respective page with correctly resolved blog ID (and in some cases, download/backup/restore ID).

- `jetpack.cloud.localhost:3000/dashboard`
- `jetpack.cloud.localhost:3000/backups/{siteSlug}`
- `jetpack.cloud.localhost:3000/backups/{siteSlug}/details/12345678`
- `jetpack.cloud.localhost:3000/backups/{siteSlug}/download/12345678`
- `jetpack.cloud.localhost:3000/backups/{siteSlug}/restore/123455678`
- `jetpack.cloud.localhost:3000/scan/{siteSlug}`
- `jetpack.cloud.localhost:3000/scan/{siteSlug}/history`
- `jetpack.cloud.localhost:3000/settings/{siteSlug}`

Ensure there are no issues in the build, no errors in the console, and no effect when booting normal Calypso.

### CODE REVIEW

File: client/landing/jetpack-cloud/routes.js [hunk 1]
```diff
@@ -6,24 +6,36 @@ import page from 'page';
...
 const router = () => {
 	page( '*', normalize );
 
-	page( '/', setupSidebar, jetpackCloud, makeLayout, clientRender );
+	page( '/', setupSidebar, jetpackCloudDashboard, makeLayout, clientRender );
 
-	page( '/backups', setupSidebar, jetpackBackups, makeLayout, clientRender );
+	page( '/backups', siteSelection, setupSidebar, jetpackCloudBackups, makeLayout, clientRender );
+	page( '/backups/:site', siteSelection, setupSidebar, jetpackCloudBackups, makeLayout, clientRender );
+	page( '/backups/:site/detail/:backupId', siteSelection, setupSidebar, jetpackCloudBackupDetail, makeLayout, clientRender );
```
Comment (delawski) [inline client/landing/jetpack-cloud/routes.js]: What do you think about having the `:site` part after the subpage, i.e.:

```
/backups/detail/:site/:backupId
```

I don't have any strong opinion on either approach. However, it seems that on wordpress.com we are using a structure like this as well:

```
/me/purchases/:site/:purchaseId
/stats/month/:site?startDate=2019-7-1
```

Probably one of the advantages of such a structure is a slightly easier way of defining links. E.g. in the sidebar we would have something like this:

```js
link={ '/backups/restore' + this.props.siteSuffix }
```

This is obviously not a blocker of any kind, just something worth a second thought. What is your opinion?

File: client/landing/jetpack-cloud/sections/backup-detail/controller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Internal dependencies
+ */
+import JetpackCloudBackupDetailPage from './main';
+
+export function jetpackCloudBackupDetail( context, next ) {
+	const backupId = parseInt( context.params.backupId );
+
+	context.primary = <JetpackCloudBackupDetailPage backupId={ backupId } />
```
Comment (delawski) [inline client/landing/jetpack-cloud/sections/backup-detail/controller.js]:
```suggestion
	context.primary = <JetpackCloudBackupDetailPage backupId={ backupId } />;
```

File: client/landing/jetpack-cloud/sections/backup-restore/controller.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import JetpackCloudBackupRestorePage from './main';
+
+export function jetpackCloudBackupRestore( context, next ) {
+	const restoreId = parseInt( context.params.restoreId );
+
+	context.primary = (
+		<JetpackCloudBackupRestorePage
+			restoreId={
+				context.params.restoreId
+					? parseInt( context.params.restoreId )
```
Comment (delawski) [inline client/landing/jetpack-cloud/sections/backup-restore/controller.js]:
```suggestion
					? restoreId
```

File: client/landing/jetpack-cloud/controller.js [hunk 4]
```diff
@@ -11,7 +11,6 @@ import { Provider as ReduxProvider } from 'react-redux';
 import JetpackCloudLayout from './layout';
 import JetpackCloudSidebar from './components/sidebar';
 import LogItem from './components/log-item';
```
Comment (delawski) [inline client/landing/jetpack-cloud/controller.js]: We don't use it here anymore.

```suggestion
```

File: client/landing/jetpack-cloud/routes.js [hunk 5]
```diff
@@ -6,24 +6,36 @@ import page from 'page';
...
+	page( '/backups/:site/download/:downloadId', siteSelection, setupSidebar, jetpackCloudBackupDownload, makeLayout, clientRender )
```
Comment (delawski) [inline client/landing/jetpack-cloud/routes.js]:
```suggestion
	page( '/backups/:site/download/:downloadId', siteSelection, setupSidebar, jetpackCloudBackupDownload, makeLayout, clientRender );
```

File: client/landing/jetpack-cloud/routes.js [hunk 6]
```diff
@@ -6,24 +6,36 @@ import page from 'page';
...
+	page( '/backups/:site/restore', siteSelection, setupSidebar, jetpackCloudBackupRestore, makeLayout, clientRender )
```
Comment (delawski) [inline client/landing/jetpack-cloud/routes.js]:
```suggestion
	page( '/backups/:site/restore', siteSelection, setupSidebar, jetpackCloudBackupRestore, makeLayout, clientRender );
```

File: client/landing/jetpack-cloud/routes.js (hunk 1 above)
Comment (rcoll) [reply client/landing/jetpack-cloud/routes.js]: Interesting! I have no strong opinion, only looking to follow convention. I'll consider changing it. Thanks!

**Pull request merged**